pkgs/by-name/zi/zigbee2mqtt/package.nix

```diff
@@ -1,38 +1,60 @@
 {
   lib,
   stdenv,
-  buildNpmPackage,
   fetchFromGitHub,
+  nodejs,
+  npmHooks,
+  pnpm_9,
   systemdMinimal,
   nixosTests,
   nix-update-script,
   withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemdMinimal,
 }:
 
-buildNpmPackage rec {
+let
+  pnpm = pnpm_9;
+in
+stdenv.mkDerivation (finalAttrs: {
   pname = "zigbee2mqtt";
-  version = "1.42.0";
+  version = "2.0.0";
 
   src = fetchFromGitHub {
     owner = "Koenkk";
     repo = "zigbee2mqtt";
-    rev = version;
-    hash = "sha256-/7mZrf3FyIliCzsy6yzVRJYMy4bViphYi81UY43iO98=";
+    tag = finalAttrs.version;
+    hash = "sha256-RHIf1m58bvSg2vezYVqStmTOOdqyto6N0hAX7sdQzFY=";
   };
 
-  npmDepsHash = "sha256-heqTYLC+TQPQ2dc5MrVdvJeNqrygC4tUgkLcfKvlYvE=";
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-fijMAt6nSbXhrm1gougcVOF1VGB3EiwnpbTE4wxy508=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    npmHooks.npmInstallHook
+    pnpm.configHook
+  ];
 
   buildInputs = lib.optionals withSystemd [
     systemdMinimal
   ];
 
-  npmFlags = lib.optionals (!withSystemd) [ "--omit=optional" ];
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm run build
+
+    runHook postBuild
+  '';
+
+  dontNpmPrune = true;
 
   passthru.tests.zigbee2mqtt = nixosTests.zigbee2mqtt;
   passthru.updateScript = nix-update-script { };
 
   meta = with lib; {
```

Comment:
remove with lib

Reply:
Hi there, I recommend stopping making those suggestions. Efforts are really burned by doing pointless work of removing what boils down to taste most of the time If you really care about `with lib;`, consider writing automation that will take care of this automatically and enforce it in CI.

Reply:
Ok

Reply:
> Hi there, I recommend stopping making those suggestions. Efforts are really burned by doing pointless work of removing what boils down to taste most of the time If you really care about `with lib;`, consider writing automation that will take care of this automatically and enforce it in CI.

I was cooking this PR this afternoon. It still has some rough edges like not detecting some breakages because of with. I'm going to continue it tomorrow.

https://github.com/NixOS/nixpkgs/pull/371665